### ISSUE TITLE
disable non-chrome-extension origins from navigating to chrome-extension

### ISSUE THREAD

**Issue by diracdeltas** [brave/browser-laptop#14772]
independent of #14712 , we should generally not allow origins that are not chrome-extension to navigate to chrome-extension

note it's already blocked in some origins like http and https

STR:
1. save the following to a local file
```
<html>
    <head>
        <meta http-equiv="refresh" content="0;url=chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/about-preferences.html" />
        <title>Page Moved</title>
    </head>
    <body>
        This page has moved. 
    </body>
</html>
```
2. open it in brave
3. notice that it navigates to about:preferences

**Comment by diracdeltas** [comment]
@bsclifton i think this is now in the hotfix since https://github.com/brave/muon/pull/635 was merged into 8.0.6.
https://github.com/brave/muon/commits/v8.0.6

**Comment by diracdeltas** [comment]
verified this behavior is fixed in 0.23.77

**Comment by srirambv** [comment]
Verified on Windows using 
* 0.23.77 0125b5f
* Muon	`8.0.6`
*libchromiumcontent	`68.0.3440.84`
